api/server/call_logs.go

```diff
@@ -21,18 +41,29 @@ func (s *Server) handleCallLogGet(c *gin.Context) {
 		return
 	}
 
-	// TODO this API needs to change to text/plain / gzip anyway, punting
-	// optimization, but we can write this direct to the wire, too... seems like
-	// we should write some kind of writev json thing for go since we keep
-	// hitting this :(
-	var b bytes.Buffer
-	b.ReadFrom(logReader)
+	mimeTypes, _ := c.Request.Header["Accept"]
 
-	callObj := models.CallLog{
-		CallID:  callID,
-		AppName: appName,
-		Log:     b.String(),
+	if len(mimeTypes) == 0 {
+		writeJSON(c, callID, appName, logReader)
+		return
 	}
 
-	c.JSON(http.StatusOK, callLogResponse{"Successfully loaded log", &callObj})
+	for _, mimeType := range mimeTypes {
+		if strings.Contains(mimeType, "application/json") {
+			writeJSON(c, callID, appName, logReader)
+			return
+		}
+		if strings.Contains(mimeType, "text/plain") {
+			io.Copy(c.Writer, logReader)
+			return
+
+		}
+		if strings.Contains(mimeType, "*/*") {
+			writeJSON(c, callID, appName, logReader)
+			return
+		}
+	}
+	// if we've reached this point it means that Fn didn't recognize Accepted content type
+	WriteError(c, c.Writer, http.StatusNotAcceptable,
```

Comment:
every other call in the front end uses `handleErrorResponse`, with an error that implements `models.APIError`. please do the same here, so that we might log this in `handleErrorResponse`, add a metric for it, whatever else we may want to do in there so that we have telemetry around this and uniformity in the front end error handling.